### DISCUSSION TITLE
ARM64 docker images + Non-Proprietary fonts for UI

### DISCUSSION THREAD

**Discussion by zhangguiyu** [PrefectHQ/prefect#4475] [Q&A]
Dear Sir/Madame,

I am trying to deploy prefect to my ARM64 Kubernetes cluster, but encounter problems when re-compiling the docker images for ARM64 for

1. ui : requires professional license to awesomefont.com, which cost $99/year

2. apollo: cannot find this on github/prefecthq, is this proprietary?

Many thanks for your help. I don't need you to provide ARM64 version of the images, I can recompile them but is it possible to:

1. release a version of ui that does not require proprietary fonts?
2. release the source code of apollo in github?

Julian

**Comment by zanieb** [comment]
Hi @zhangguiyu

1. I can't comment on the necessity of the fonts but I can reach out to our UI team. Perhaps we can build an arm64 UI image with the fonts included.

2. Our Apollo service is at https://github.com/PrefectHQ/server/blob/master/services/apollo

**Reply by zhangguiyu** [reply]
Wow! 2 hours resposne time! Thanks for the link to apollo, I will try to compile it.

**Comment by znicholasbrown** [comment]
Hi @zhangguiyu - thanks for opening this discussion. At this time we can't commit to releasing a version of the UI with a completely different set of iconography (though I won't rule this out for the future). However, you could work around this by copying the files that live in the `/var/www` directory of an already-compiled container to your arm64 compliant one.

**Reply by zhangguiyu** [reply]
Thanks for the pointer! Let me try to see if I can extract the fonts and/or token-key from /var/www of a compiled image.

Amazing response time!

**Reply by zhangguiyu** [reply]
hmm, I saw various fonts in a pre-built x86 UI.

Moreover, I cannot build the ARM64 image of UI because most places require an FA_TOKEN (fontawesome)

`[ec2-user@ip-192-168-222-122 ui]$ ./build
Sending build context to Docker daemon  13.02MB
Step 1/20 : FROM node:14.7.0 as ui
 ---> 1708ae512ffe
Step 2/20 : ARG PREFECT_VERSION=development
 ---> Running in 3bd1c0854de3
Removing intermediate container 3bd1c0854de3
 ---> 9dd06aa4f44f
Step 3/20 : ENV PREFECT_VERSION=$PREFECT_VERSION
 ---> Running in fbfaab493c48
Removing intermediate container fbfaab493c48
 ---> d6a8c7f5f542
Step 4/20 : RUN echo "VUE_APP_PREFECT_VERSION=${PREFECT_VERSION}" >> .env
 ---> Running in 3940d939fce8
Removing intermediate container 3940d939fce8
 ---> 6804b6db767e
Step 5/20 : COPY ./ /app
 ---> 6fab7e29fafb
Step 6/20 : COPY ./LICENSE LICENSE
 ---> ee8cf93e0303
Step 7/20 : WORKDIR /app
 ---> Running in 396ca6f4d056
Removing intermediate container 396ca6f4d056
 ---> 797353d400ae
Step 8/20 : RUN --mount=type=secret,id=FA_TOKEN cp /run/secrets/FA_TOKEN .npmrc
 ---> Running in 7be5a947ade1
cp: cannot stat '/run/secrets/FA_TOKEN': No such file or directory
`
